src/citrine/informatics/data_sources.py

```diff
@@ -74,6 +104,20 @@ def __init__(self,
         self.column_definitions = column_definitions
         self.identifiers = identifiers
 
+    @classmethod
+    def _data_source_id_builder(cls, *args) -> DataSource:
+        # TODO Figure out how to populate the column definitions
+        return CSVDataSource(
+            file_link=FileLink(url=args[0], filename=args[1]),
+            column_definitions={}
+        )
```

Comment:
There is no obvious method to identify what the column definitions and identifiers would be from just the DataSourceId.

Reply:
Would it be too disruptive to issue a warning here, so the user knows that the data source they're getting is incomplete? Actually, maybe the warning should come when they access the `DesignWorkflow.data_source` field, if it's a `CSVDataSource` with no `column_definition`.

Either way, I'd like if we can alert them this isn't a completely accurate representation.

Reply:
Yeah, using a warning makes sense.  I don't think people are using CSVDataSources anymore, but...